src/v0/destinations/pinterest_tag/utils.js

```diff
@@ -30,6 +30,41 @@ const ecomEventMaps = [
 
 const USER_NON_ARRAY_PROPERTIES = ['client_user_agent', 'client_ip_address'];
 
+const getHashedValue = (key, value) => {
+  switch (key) {
+    case 'em':
+    case 'ct':
+    case 'st':
+    case 'country':
+    case 'ln':
+    case 'fn':
+    case 'ge':
+      value = Array.isArray(value)
+        ? value.map((val) => val.toString().toLowerCase())
+        : value.toString().toLowerCase();
+      break;
+    case 'ph':
+      // phone numbers should only contain digits & should not contain leading zeros
+      value = Array.isArray(value)
+        ? value.map((val) => val.toString().replace(/\D/g, '').replace(/^0+/, ''))
+        : value.toString().replace(/\D/g, '').replace(/^0+/, '');
+      break;
+    case 'zp':
+      // zip fields should only contain digits
+      value = Array.isArray(value)
+        ? value.map((val) => val.toString().replace(/\D/g, ''))
+        : value.toString().replace(/\D/g, '');
+      break;
+    case 'hashed_maids':
+    case 'external_id':
+      // no action needed on value
+      break;
+    default:
+      return String(value);
```

Comment:
For the other `keys`, Is hashing not required ?

Reply:
I see that's the existing production behaviour

Reply:
Yes, that is not required